sdk/core/src/policy.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{Request, Response};
+
+use async_trait::async_trait;
+
+use std::error::Error;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+
+pub type PolicyResult<T> = Result<T, Box<dyn Error + Send + Sync>>;
+
+#[derive(Copy, Clone, Debug)]
+pub struct RetryOptions {
+    num_retries: usize,
```

Comment:
At some point we should respect headers sent by the server, for e.g. in .NET (probably same across languages):

```csharp
        private const string RetryAfterHeaderName = "Retry-After";
        private const string RetryAfterMsHeaderName = "retry-after-ms";
        private const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
```

See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs.

Reply:
BTW: There's a lot of good policies in that folder we should eventually port. Most SDK languages have these or something like them.

Reply:
Definitely, the policies as they are, are really just proofs of concept. We'll need to improve them considerably.